Tests/UnitTests/Plugins/Detray/DetrayMaterialConverterTests.cpp

```diff
@@ -0,0 +1,348 @@
+  // Thickness should be set to one
+  CHECK_CLOSE_ABS(payload.thickness, 1.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  // Index in collection not set at this simple conversion
+  BOOST_CHECK(!payload.index_in_coll.has_value());
+  // Material parameters in detray are (x0, l0, ar, z, mass_density,
+  // molar_density, solid/liquid/etc. flag ... ignored currently)
+  CHECK_CLOSE_ABS(payload.mat.params[0u], 1.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.mat.params[1u], 2.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.mat.params[2u], 3.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.mat.params[3u], 4.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  BOOST_CHECK_NE(payload.mat.params[4u], payload.mat.params[5u]);
+  CHECK_CLOSE_ABS(payload.mat.params[5u], 5.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.mat.params[6u], 0.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+}
+
+BOOST_AUTO_TEST_CASE(DetrayBinnedMaterialConversionX) {
+  // Create a binned material in 4 bins in x direction
+  Acts::BinUtility binUtility(4u, -2., 2., Acts::BinningOption::open,
+                              Acts::BinningValue::binX);
+
+  std::vector<Acts::MaterialSlab> materialSlabs = {
+      materialSlab12345, materialSlab678910, materialSlab54321,
+      materialSlab109876};
+
+  auto binnedMaterial = Acts::BinnedSurfaceMaterial(
+      binUtility, {materialSlabs}, 0., Acts::MappingType::Default);
+
+  detray::io::grid_payload<detray::io::material_slab_payload,
+                           detray::io::material_id>
+      payload = Acts::DetrayMaterialConverter::convertSurfaceMaterial(
+          binnedMaterial, *logger);
+  // Check the payload
+  // - we fake 2D always for detray to minimize the number of containers
+  // - four material payloads in the grid
+  BOOST_CHECK_EQUAL(payload.axes.size(), 2u);
+  BOOST_CHECK_EQUAL(payload.bins.size(), 4u);
+  BOOST_CHECK(!payload.transform.has_value());
+  BOOST_CHECK(payload.grid_link.type ==
+              detray::io::material_id::rectangle2_map);
+  // x-axis
+  BOOST_CHECK(payload.axes[0u].label == detray::axis::label::e_x);
+  BOOST_CHECK_EQUAL(payload.axes[0u].bins, 4u);
+  CHECK_CLOSE_ABS(payload.axes[0u].edges[0u], -2.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  CHECK_CLOSE_ABS(payload.axes[0u].edges[1u], 2.,
+                  std::numeric_limits<Acts::ActsScalar>::epsilon());
+  BOOST_CHECK(payload.axes[0u].binning == detray::axis::binning::e_regular);
+  BOOST_CHECK(payload.axes[0u].bounds == detray::axis::bounds::e_closed);
+  // axis is dummy
+  BOOST_CHECK(payload.axes[1u].label == detray::axis::label::e_y);
+  BOOST_CHECK_EQUAL(payload.axes[1u].bins, 1u);
```

Comment:
Will you test somewhere that the local bin indices are assigned correctly to the slab payloads?

Reply:
Let's do it.